src/components/molecules/VenueChat/VenueChat.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useContext, useEffect, useState, useMemo, FC } from "react";
+import { ChatContext } from "components/context/ChatContext";
+import "./CampChat.scss";
+import { useUser } from "hooks/useUser";
+import { useSelector } from "hooks/useSelector";
+import { useVenueId } from "hooks/useVenueId";
+import { getDaysAgoInSeconds } from "utils/time";
+import { VENUE_CHAT_AGE_DAYS } from "settings";
+import ChatBox from "components/molecules/Chatbox";
+import useRoles from "hooks/useRoles";
+
+interface ChatOutDataType {
+  messageToTheBand: string;
+}
+
+const VenueChat: FC = () => {
+  const venueId = useVenueId();
+  const { userRoles } = useRoles();
+  const { user } = useUser();
+  const { chats, venue } = useSelector((state) => ({
+    chats: state.firestore.ordered.venueChats,
+    venue: state.firestore.data.currentVenue,
+  }));
+
+  const [isMessageToTheBarSent, setIsMessageToTheBarSent] = useState(false);
+
+  useEffect(() => {
+    if (isMessageToTheBarSent) {
+      setTimeout(() => {
+        setIsMessageToTheBarSent(false);
+      }, 2000);
+    }
+  }, [isMessageToTheBarSent]);
+
+  const chatContext = useContext(ChatContext);
+
+  const onMessageSubmit = async (data: ChatOutDataType) => {
+    chatContext &&
+      user &&
+      chatContext.sendRoomChat(user.uid, venueId!, data.messageToTheBand);
+  };
```

Comment:
As above RE: `useCallback` + semantically naming functions by their 'action'/'effect'

Reply:
fixed in 77fc5d6d1829a42da2152d00871cde0b6f47269d